### PR TITLE
NullReferenceException when setting BarBackgroundColor for a NavigationPage - fix

### LINKED ISSUE #25114: NullReferenceException when setting BarBackgroundColor for a NavigationPage
### Description

From 8.0.90 (as far as I can tell) I get NRE when attempting to set a NavigationPage's BarBackgroundColor.
I'm not sure of how this is failing (ie I can't share a repro project at the moment), but it can't hurt to get the ball rolling on this topic, perhaps others have had similar issues?

this is running on a `Flyout` page - its detail is set to a navigation page that encapsulates a content page.
when the content page triggers `OnAppearing` we set the BarBackgroundColor for the navigation page.

however, when it gets to the `NavigationPageToolbar.ApplyChanges` method, it blows up, as it seemingly receives a null navigationPage object

image

and that null object comes from the call from `NavigationPageToolbar.OnPagePropertyChanged`
image

Microsoft.Maui.Controls.dll!Microsoft.Maui.Controls.NavigationPageToolbar.ApplyChanges(Microsoft.Maui.Controls.NavigationPage navigationPage)	Unknown
 	Microsoft.Maui.Controls.dll!Microsoft.Maui.Controls.NavigationPageToolbar.OnPagePropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)	Unknown
 	Microsoft.Maui.Controls.dll!Microsoft.Maui.Controls.MenuItemTracker<System.__Canon>.OnPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs propertyChangedEventArgs)	Unknown
 	Microsoft.Maui.Controls.dll!Microsoft.Maui.Controls.BindableObject.OnPropertyChanged(string propertyName)	Unknown
 	Microsoft.Maui.Controls.dll!Microsoft.Maui.Controls.Element.OnPropertyChanged(string propertyName)	Unknown
 	Microsoft.Maui.Controls.dll!Microsoft.Maui.Controls.BindableObject.SetValueActual(Microsoft.Maui.Controls.BindableProperty property, Microsoft.Maui.Controls.BindableObject.BindablePropertyContext context, object value, bool currentlyApplying, Microsoft.Maui.Controls.Internals.SetValueFlags attributes, Microsoft.Maui.Controls.SetterSpecificity specificity, bool silent)	Unknown
 	Microsoft.Maui.Controls.dll!Microsoft.Maui.Controls.BindableObject.SetValueCore(Microsoft.Maui.Controls.BindableProperty property, object value, Microsoft.Maui.Controls.Internals.SetValueFlags attributes, Microsoft.Maui.Controls.BindableObject.SetValuePrivateFlags privateAttributes, Microsoft.Maui.Controls.SetterSpecificity specificity)	Unknown
 	Microsoft.Maui.Controls.dll!Microsoft.Maui.Controls.BindableObject.SetValue(Microsoft.Maui.Controls.BindableProperty property, object value)	Unknown
 	Microsoft.Maui.Controls.dll!Microsoft.Maui.Controls.NavigationPage.BarBackgroundColor.set(Microsoft.Maui.Graphics.Color value)	Unknown

### Steps to Reproduce

_No response_

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.91 SR9.1

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.82 SR8.2

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/25114
Code from this repo https://github.com/markmccaigue/MAUI8090BarBackgroundColorCrash/tree/main

|iOS|Android|
|--|--|
|<video src="https://github.com/user-attachments/assets/44b3376e-752e-4e4a-a91b-8bb7062c93eb" width="300px"/>|<video src="https://github.com/user-attachments/assets/17fcb331-c685-4a21-8d98-6c65fb077514" width="300px"/>|

### CODE REVIEW

**Review by Redth — DISMISSED** [review]

**Review by jsuarezruiz — DISMISSED** [review]

**Review by PureWeen — CHANGES_REQUESTED** [review]
Let's add UITest if we can 

Also, I'm curious why the code path is reaching here in the first place when it didn't used to

File: src/Controls/tests/TestCases.HostApp/Issues/Issue25114.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+namespace Maui.Controls.Sample.Issues;
```
Comment (PureWeen) [inline src/Controls/tests/TestCases.HostApp/Issues/Issue25114.xaml.cs:1]: Can you remove the ".xaml." extension from this file?

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**